### PR TITLE
[NETBEANS-862] Added auto-complete support for var keyword in JDK-11 …

### PR DESCRIPTION
**Pull request by arusinha**
…Lambda Parameters
Auto-complete(CTL+ space)  for var keyword in not currently supported in Lambda parameter (JDK-11). Cursor is at caret position.
Added support for that.

IntBinaryOperator calc2 = (var x,  ^ y ) -> x+y;
Jira Link:
https://issues.apache.org/jira/browse/NETBEANS-862

### CODE REVIEW

File: java/java.completion/test/unit/data/org/netbeans/modules/java/completion/data/SimpleLambdaExpression2Start.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    private void test(Foo f) {
+    }
+    
+    private void test2(String str, int num, Foo f) {
+    }
+    
+    public static void main(String[] args) {
+	Test t = new Test();
```
Comment (rtaneja1) [inline java/java.completion/test/unit/data/org/netbeans/modules/java/completion/data/SimpleLambdaExpression2Start.java]: Closing bracket missing?

File: java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java [hunk 2]
```diff
@@ -5869,4 +5899,40 @@ private boolean withinBounds(Env env, TypeMirror type, List<? extends TypeMirror
         }
         return true;
     }
+    private void addVarTypeForLambdaParam(final Env env) throws IOException {
+        if (SOURCE_VERSION_RELEASE_11 == null || env.getController().getSourceVersion().compareTo(SOURCE_VERSION_RELEASE_11) < 0) {
+            return;
+        }
+        results.add(itemFactory.createKeywordItem(VAR_KEYWORD, SPACE, anchorOffset, false));
+    }
```
Comment (rtaneja1) [inline java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java]: Please add a line between methods and correct indentation.

File: java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java [hunk 3]
```diff
@@ -5869,4 +5899,40 @@ private boolean withinBounds(Env env, TypeMirror type, List<? extends TypeMirror
...
+/**
+ *
+ * @param env : env
+ * @param tree : Lambda expression tree
+ * @return True if first param of lambda expr is of 'var' type
+ */
+   private boolean isLambdaVarType(Env env,Tree tree) {
```
Comment (rtaneja1) [inline java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java]: Auto format newly added code.

File: java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java [hunk 4]
```diff
@@ -5869,4 +5899,40 @@ private boolean withinBounds(Env env, TypeMirror type, List<? extends TypeMirror
...
+
+        boolean isFirstParamVarType = false;
+
+        VariableTree firstParamTree = let.getParameters().get(0);
+        int firstParamStartPos = (int) env.getSourcePositions().getStartPosition(env.getRoot(), firstParamTree);
+        TokenSequence<JavaTokenId> ts = findLastNonWhitespaceToken(env, let, env.getOffset());
+        ts.move(firstParamStartPos);
+        ts.movePrevious();
+
+        // TreeUtilities.isVarType() API can't we used as not able to extract tokens from firstParamTree.
```
Comment (rtaneja1) [inline java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java]: Comment doesn't indicate the issue

File: java/java.completion/test/unit/data/org/netbeans/modules/java/completion/data/SimpleLambdaExpression2Start.java (hunk 1 above)
Comment (arusinha) [reply java/java.completion/test/unit/data/org/netbeans/modules/java/completion/data/SimpleLambdaExpression2Start.java]: It was deliberately done as I want to add code inside main method. If I add closing bracket then test code will be added after the main method which I don't want

**Comment by vikasprabhakar** [comment]
👍

**Pull request merged**